x-pack/plugins/security_solution/server/endpoint/routes/agent/agent_status_handler.ts

```diff
@@ -79,12 +81,13 @@ export const getAgentStatusRouteHandler = (
       esClient,
       soClient,
       endpointService: endpointContext.service,
+      connectorActionsClient: (await context.actions).getActionsClient(),
```

Comment:
If you make the `connectorActionsClient` as optional or that it takes `undefined` (see my later comment) as we do not need it for SentinelOne agent status

```suggestion
      connectorActionsClient:
        agentType === 'crowdstrike' ? (await context.actions).getActionsClient() : undefined,
```